sources/packages/web/src/services/http/ApplicationApi.ts

```diff
@@ -146,36 +151,72 @@ export class ApplicationApi extends HttpBaseClient {
   /**
    * API Client for application detail.
    * @param applicationId
-   * @param userId
    * @returns
    */
   public async getApplicationDetails(
     applicationId: number,
-    studentId: number,
   ): Promise<GetApplicationBaseDTO> {
     const response = await this.getCall(
-      `application/${applicationId}/student/${studentId}/aest`,
+      this.addClientRoot(`application/${applicationId}`),
     );
     return response.data as GetApplicationBaseDTO;
   }
 
   /**
    * API Client to get student applications.
-   * ! Because of code duplication, this function
-   * ! is used in both AEST(Ministry) student application summary
-   * ! as well as student application summary.
-   * ! only passed URL value will be different.
-   * ! Both are using same interface
-   * ! In future, if any of them needs a
-   * ! different interface, use create a
-   * ! different functions for both
-   * @param url to be send
-   * @returns
+   * @param page, page number if nothing is passed then
+   * DEFAULT_PAGE_NUMBER is taken
+   * @param pageLimit, limit of the page if nothing is
+   * passed then DEFAULT_PAGE_LIMIT is taken
+   * @param sortField, field to be sorted
+   * @param sortOrder, order to be sorted
+   * @returns StudentApplicationAndCount
    */
-  public async getAllApplicationAndCount(
-    url: string,
+  public async getAllApplicationAndCountForStudent(
+    page = DEFAULT_PAGE_NUMBER,
+    pageCount = DEFAULT_PAGE_LIMIT,
+    sortField?: StudentApplicationFields,
+    sortOrder?: DataTableSortOrder,
   ): Promise<StudentApplicationAndCount> {
-    const response = await this.getCall(url);
-    return response.data as StudentApplicationAndCount;
+    let URL = `students/application-summary?page=${page}&pageLimit=${pageCount}`;
```

Comment:
We have some helpers in place to create the URL with paginations options, right?

Reply:
ya, I was planning to do it. missed it. Thanks